app/models/concerns/labware_creators/support_v2_source_plate.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# TODO: {Y24-190} Remove this module once the V1 API is no longer used.
+
+# Can be included in labware creators which require access to a V2 source plate when passing a V1 parent plate
+# to methods/classes requiring V2 resources.
+module LabwareCreators::SupportV2SourcePlate
+  extend ActiveSupport::Concern
+
+  # parent is using SS v1 API
+  # so this method is used to access the plate via SS v2 API
+  def source_plate
+    if parent.is_a? Limber::Plate
+      @source_plate ||= Sequencescape::Api::V2::Plate.find_by(uuid: parent.uuid) if @source_plate&.uuid != parent.uuid
+    else
+      @source_plate = nil
+    end
+
+    @source_plate
```

Comment:
overriding the getter for the instance variable to fetch the plate using v2?

Reply:
It's a pattern called memoization where you use the `@` variable to store a cached value to avoid having to perform the lookup repeatedly for the parent UUID.  Rubocop requires that you use the same name for the `@` variable as you do for the method that populates it.  See a blog article on memoization here: https://blog.appsignal.com/2022/12/20/a-guide-to-memoization-in-ruby.html

When you call `self.source_plate` from elsewhere in the class (and publically in this case), the method will be preferred over the `@` variable.

Reply:
Ah ok cool, thanks I didn't realise you could do it this way. I think I've used a similar way before something like: 
```
def source_plate
   @ source_plate ||= <method_call>
end
```

Reply:
I think that's what I'm doing here as well.  It's just complicated by the fact that we have to check the type first before converting it and an `if` check to ensure we do over-write our memoized version if, somehow, the parent has changed.